bindings/muen/muen-yield.c

```diff
@@ -18,23 +18,25 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include "../bindings.h"
+#include "bindings.h"
 #include "muen-net.h"
 
-bool solo5_yield(uint64_t deadline)
+void solo5_yield(solo5_time_t deadline, solo5_handle_set_t *ready_set)
 {
-    bool rc = false;
+    solo5_handle_set_t tmp_ready_set = 0;
+
     do {
-        if (muen_net_pending_data()) {
-            rc = true;
-            break;
+        for (solo5_handle_t i = 0U; i < MFT_MAX_ENTRIES; ++i) {
+            if (muen_net_pending_data(i))
+                tmp_ready_set |= 1UL << i;
```

Comment:
While this is a neat trick, this only works when `MFT_MAX_ENTRIES <= sizeof(tmp_ready_set)`.
I am a little bit skeptical about hardcoding that invariant in too many places without either statically checking the `MFT_MAX_ENTRIES` constant or the ABI version, WDYT @mato?

Reply:
@cfcs We could do something like (not tested)

```
_Static_assert(MFT_MAX_ENTRIES <= (sizeof(solo5_ready_set_t) * 8), "MFT_MAX_ENTRIES too large")
```

...but I don't know where to put that, as it depends on _both_ `mft_abi.h` and `solo5.h`, the two of which are supposed to be independent.